### PR TITLE
Feature user

### PR DESCRIPTION
**Pull request by seamile**

### CODE REVIEW

**Comment by binyoucai** [comment]
ok ok

**Comment by Seven5y** [comment]
not ok

File: .gitignore [hunk 1]
```diff
@@ -14,7 +14,6 @@ dist/
 downloads/
```
Comment (nbfccsdwll) [inline .gitignore]: OK

**Comment by SceneryCN** [comment]
nine sun baby

**Comment by Seven5y** [comment]
手滑手滑

**Comment by TheodoreLv** [comment]
我同意啦!

File: common/errors.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    def __str__(self):
+        return self.__class__.__name__
+
+
+def generate_logic_error(name: str, code: int) -> LogicError:
+    '''创建逻辑异常的子类'''
+    base_cls = (LogicError,)
+    return type(name, base_cls, {'code': code})
+
```
Comment (fanfanstl) [inline common/errors.py]: sunck

**Pull request merged**